core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -194,11 +194,68 @@ object DataMirror {
     }
   }
 
-  /** Return all expanded components, including intermediate aggregate nodes
+  // Old definition of collectLeafMembers
+  @deprecated("Use DataMirror.collectLeafMembers instead")
+  def getLeafs(d: Data): Seq[Data] = collectLeafMembers(d)
+
+  // Old definition of collectAllChildren
+  @deprecated("Use DataMirror.collectAllMembers instead")
+  def getIntermediateAndLeafs(d: Data): Seq[Data] = collectAllMembers(d)
+
+  /** Recursively collect just the leaf components of a data component's children
+    * (i.e. anything that isn't a `Record` or a `Vec`, but an `Element`)
+    *
+    * @param d Data component to recursively collect leaf components.
+    *
+    * @return All `Element` components; intermediate fields/indices are not included
+    */
+  def collectLeafMembers(d: Data): Seq[Data] =
+    DataMirror.collectMembers(d) { case x: Element => x.asInstanceOf[Data] }.toVector
```

Comment:
```suggestion
    DataMirror.collectMembers[Data](d) { case x: Element => x }.toVector
```
Is that `.asInstanceOf` necessary? If you're getting a type error, try either the above or below:
```suggestion
    DataMirror.collectMembers(d) { case x: Element => x: Data }.toVector
```

This is a small thing, but that cast does have a runtime cost and these static casts might be able to elide it.

Reply:
```suggestion
    DataMirror.collectMembers(d) { case x: Element => x }.toVector
```
Looks like this will work.